CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/resourcetypes/CVKImage.java

```diff
@@ -455,28 +600,353 @@ public static CVKImage Create(  CVKDevice cvkDevice,
             }        
 
             vkBindImageMemory(cvkDevice.GetDevice(), cvkImage.pImage.get(0), cvkImage.pImageMemory.get(0), 0);
-            
-            // Create an image view that can be used to read and write this image
-            VkImageViewCreateInfo vkViewInfo = VkImageViewCreateInfo.callocStack(stack);
-            vkViewInfo.sType(VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO);
-            vkViewInfo.image(cvkImage.GetImageHandle());
-            vkViewInfo.viewType(cvkImage.viewType);
-            vkViewInfo.format(cvkImage.GetFormat());
-            vkViewInfo.subresourceRange().aspectMask(cvkImage.GetAspectMask());
-            vkViewInfo.subresourceRange().baseMipLevel(0);
-            vkViewInfo.subresourceRange().levelCount(1);
-            vkViewInfo.subresourceRange().baseArrayLayer(0);
-            vkViewInfo.subresourceRange().layerCount(layers);
-
-            ret = vkCreateImageView(cvkDevice.GetDevice(), vkViewInfo, null, cvkImage.pImageView);
-            checkVKret(ret);           
-            
-            return cvkImage;
+        }
+        return cvkImage;
+    }
+
+
+    /**
+     * Creates an ImageView associate with the image handle
+     * @return error code
+     */
+    public int CreateImageView() {
+        int ret = VK_SUCCESS;
+        
+        try(MemoryStack stack = stackPush()) {
+                // Create an image view that can be used to read and write this image
+                VkImageViewCreateInfo vkViewInfo = VkImageViewCreateInfo.callocStack(stack);
+                vkViewInfo.sType(VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO);
+                vkViewInfo.image(GetImageHandle());
+                vkViewInfo.viewType(viewType);
+                vkViewInfo.format(GetFormat());
+                vkViewInfo.subresourceRange().aspectMask(GetAspectMask());
+                vkViewInfo.subresourceRange().baseMipLevel(0);
+                vkViewInfo.subresourceRange().levelCount(1);
+                vkViewInfo.subresourceRange().baseArrayLayer(0);
+                vkViewInfo.subresourceRange().layerCount(layers);
+
+                ret = vkCreateImageView(cvkDevice.GetDevice(), vkViewInfo, null, pImageView);
+                checkVKret(ret);
+
         } catch (Exception e) {
-            //TODO_TT: move this to class destructor
-            cvkImage.Destroy();
+            // TODO HYDRA TEST THIS CODE PATH       
+            cvkDevice.GetLogger().info("Image View creation failed for Image Handle %d. Image will be destroyed!", GetImageHandle());
+            ret = CVK_ERROR_IMAGE_VIEW_CREATION_FAILED;
+            
+            DestroyImage();
+            DestroyImageMemory();
         }
         
-        return null;
+        return ret;
+    }
+    
+    
+    /**
+     * Prepares the image 
+     * 
+     * @param stack
+     * @return 
+     */
+    public CVKImage PrepareImageToMemory(MemoryStack stack) {
```

Comment:
I'm not sure what this is doing from the name.

Reply:
renamed to CopyToCPUMemoryImage()